### PR TITLE
Jonathan changes fixed

### PR DESCRIPTION
**Pull request by Haplicity**
-added editorconfig and updated resulting files
-added dist to .gitignore and removed dist folders from components
-updated package.json's scripts and added private: true
-removed package-lock.json from gitignore and re-added package-lock.json
-updated packages

### CODE REVIEW

**Comment by jonathantneal** [comment]
Hi. Happy to help, but it’s really hard to follow this in parallel with all the reverts you committed without review. I also don’t know why dist is coming back.

Could you give a summary of 1. what was working before my changes, 2. what stopped working after my changes, 3. what stopped or started working after your reverts, and 4. what this PR changes from after all those reverts?

**Comment by Haplicity** [comment]
1. npm run docz:dev was working before your changes
2. npm run docz:dev broke after your changes
3. the reverts were just reverting your changes, re-adding them, and then reverting them again. Not exactly the cleanest approach on my end, I admit.
4. See description.

**Comment by jonathantneal** [comment]
Cool. So, so this reverts the revert of my PR, then adds back `package-lock.json`, then adds `npm run build` to `postinstall`, and then adds `private` to `package.json`. I’ll give it a second review.

**Comment by jonathantneal** [comment]
I’m unable to verify the fixes due to the following error after running `npm install`:

```
npm WARN deprecated babel-preset-docz@0.13.6: this package is deprecated
npm WARN deprecated kleur@2.0.2: Please upgrade to kleur@3 or migrate to 'ansi-colors' if you prefer the old syntax. Visit <https://github.com/lukeed/kleur/releases/tag/v3.0.0\> for migration path(s).

> fsevents@1.2.7 install ~/GitHub/t7/react-ui/node_modules/fsevents
> node install

node-pre-gyp WARN Using request for node-pre-gyp https download
[fsevents] Success: "~/GitHub/t7/react-ui/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64/fse.node" is installed via remote

> husky@1.3.1 install ~/GitHub/t7/react-ui/node_modules/husky
> node husky install

husky > setting up git hooks
husky > done

> @0.0.1 postinstall ~/GitHub/t7/react-ui
> lerna run build && npm run bootstrap

lerna notice cli v3.13.1
lerna info versioning independent
lerna info Executing command in 8 packages: "npm run build"
lerna ERR! npm run build exited 1 in '@t7/paging'
lerna ERR! npm run build stdout:

> @t7/paging@0.0.3 build ~/GitHub/t7/react-ui/packages/paging
> webpack --config ../../webpack.config.js


lerna ERR! npm run build stderr:
~/GitHub/t7/react-ui/packages/paging/node_modules/webpack-cli/bin/cli.js:231
                                throw err;
                                ^

TypeError: acorn.Parser.extend is not a function
```

**Comment by jonathantneal** [comment]
Further details:

```
~/GitHub/t7/react-ui/packages/paging/node_modules/webpack-cli/bin/cli.js:231
                                throw err;
                                ^

TypeError: acorn.Parser.extend is not a function
    at Object.<anonymous> (~/GitHub/t7/react-ui/packages/paging/node_modules/webpack/lib/Parser.js:18:34)
    at Module._compile (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:178:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:810:10)
    at Module.load (internal/modules/cjs/loader.js:666:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:606:12)
    at Function.Module._load (internal/modules/cjs/loader.js:598:3)
    at Module.require (internal/modules/cjs/loader.js:705:19)
    at require (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (~/GitHub/t7/react-ui/packages/paging/node_modules/webpack/lib/JavascriptModulesPlugin.js:7:16)
    at Module._compile (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:178:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:810:10)
    at Module.load (internal/modules/cjs/loader.js:666:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:606:12)
    at Function.Module._load (internal/modules/cjs/loader.js:598:3)
    at Module.require (internal/modules/cjs/loader.js:705:19)
    at require (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (~/GitHub/t7/react-ui/packages/paging/node_modules/webpack/lib/WebpackOptionsApply.js:9:33)
    at Module._compile (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:178:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:810:10)
    at Module.load (internal/modules/cjs/loader.js:666:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:606:12)
    at Function.Module._load (internal/modules/cjs/loader.js:598:3)
    at Module.require (internal/modules/cjs/loader.js:705:19)
    at require (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (~/GitHub/t7/react-ui/packages/paging/node_modules/webpack/lib/webpack.js:10:29)
    at Module._compile (~/GitHub/t7/react-ui/packages/paging/node_modules/v8-compile-cache/v8-compile-cache.js:178:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:810:10)
    at Module.load (internal/modules/cjs/loader.js:666:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:606:12)
    at Function.Module._load (internal/modules/cjs/loader.js:598:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @t7/paging@0.0.3 build: `webpack --config ../../webpack.config.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @t7/paging@0.0.3 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2019-03-22T18_41_48_749Z-debug.log

lerna ERR! npm run build exited 1 in '@t7/paging'
lerna WARN complete Waiting for 5 child processes to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @0.0.1 postinstall: `lerna run build && npm run bootstrap`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @0.0.1 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2019-03-22T18_41_48_832Z-debug.log
```
